src/hooks/useTimeLayerData.ts

```diff
@@ -0,0 +1,220 @@
+import { useMemo } from 'react';
+
+import dayjs, { Dayjs } from 'dayjs';
+import utc from 'dayjs/plugin/utc';
+import OlLayer from 'ol/layer/Layer';
+
+import { WmsLayer } from '@terrestris/ol-util/dist/typeUtils/typeUtils';
+
+dayjs.extend(utc);
+
+export interface TimeLayerData {
+  timeAwareLayers: WmsLayer[];
+  initialTime: Dayjs | undefined;
```

Comment:
This could be simplified to `initialTime?: Dayjs;` (and also in the lines below).